gateway/api/connections/connections.go

```diff
@@ -680,3 +693,379 @@ func GetDatabaseSchemas(c *gin.Context) {
+	if script == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "unsupported database type"})
+		return
+	}
+
+	userAgent := apiutils.NormalizeUserAgent(c.Request.Header.Values)
+	client, err := clientexec.New(&clientexec.Options{
+		OrgID:          ctx.GetOrgID(),
+		ConnectionName: conn.Name,
+		BearerToken:    getAccessToken(c),
+		UserAgent:      userAgent,
+		Verb:           pb.ClientVerbPlainExec,
+	})
+	if err != nil {
+		log.Error(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+		return
+	}
+
+	respCh := make(chan *clientexec.Response)
+	go func() {
+		defer func() { close(respCh); client.Close() }()
+		select {
+		case respCh <- client.Run([]byte(script), nil):
+		default:
+		}
+	}()
+
+	timeoutCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancelFn()
+	select {
+	case outcome := <-respCh:
+		if outcome.ExitCode != 0 {
+			log.Errorf("failed issuing plain exec: %s, output=%v", outcome.String(), outcome.Output)
+			c.JSON(http.StatusInternalServerError, gin.H{"message": fmt.Sprintf("failed to get columns: %s", outcome.Output)})
+			return
+		}
+
+		response := ColumnsResponse{Columns: []openapi.ConnectionColumn{}}
+
+		if currentConnectionType == pb.ConnectionTypeMongoDB {
+			// Parse MongoDB output
+			output := cleanMongoOutput(outcome.Output)
+			if output != "" {
+				var result []map[string]interface{}
+				if err := json.Unmarshal([]byte(output), &result); err != nil {
+					log.Errorf("failed parsing mongo response: %v", err)
+					c.JSON(http.StatusInternalServerError, gin.H{"message": fmt.Sprintf("failed to parse MongoDB response: %v", err)})
+					return
+				}
+
+				for _, row := range result {
+					column := openapi.ConnectionColumn{
+						Name:     getString(row, "column_name"),
+						Type:     getString(row, "column_type"),
+						Nullable: !getBool(row, "not_null"),
+					}
+					response.Columns = append(response.Columns, column)
+				}
+			}
+		} else {
+			// Parse SQL output
+			lines := strings.Split(outcome.Output, "\n")
+
+			// Process each line (skip header)
+			startLine := 1
+			if currentConnectionType == pb.ConnectionTypeMSSQL {
+				// Find the line with dashes for MSSQL
+				for i, line := range lines {
+					if strings.Contains(line, "----") {
+						startLine = i + 1
+						break
+					}
+				}
+			}
+
+			for i, line := range lines {
+				line = strings.TrimSpace(line)
+				if i < startLine || line == "" || strings.HasPrefix(line, "(") {
+					continue
+				}
+
+				fields := strings.Split(line, "\t")
+				if len(fields) < 3 {
+					continue
+				}
+
+				column := openapi.ConnectionColumn{
+					Name:     fields[0],
+					Type:     fields[1],
+					Nullable: fields[2] != "t" && fields[2] != "1",
+				}
+				response.Columns = append(response.Columns, column)
+			}
+		}
+
+		c.JSON(http.StatusOK, response)
+
+	case <-timeoutCtx.Done():
+		client.Close()
+		log.Infof("runexec timeout (30s), it will return async")
```

Comment:
The result will not return async. Inform the it timeout to obtain the the tables columns and include the information of the connection at least along with any useful information